### PR TITLE
AP_Beacon: fix comparison order to prevent using wrong beacon_instanc…

### PR DESCRIPTION
**Pull request by khancyr**
…e first

### CODE REVIEW

File: libraries/AP_Beacon/AP_Beacon.cpp [hunk 1]
```diff
@@ -214,7 +214,7 @@ bool AP_Beacon::beacon_healthy(uint8_t beacon_instance) const
 // return distance to beacon in meters
 float AP_Beacon::beacon_distance(uint8_t beacon_instance) const
 {
-    if (!beacon_state[beacon_instance].healthy || beacon_instance >= num_beacons) {
+    if ( beacon_instance >= num_beacons || !beacon_state[beacon_instance].healthy) {
```
Comment (peterbarker) [inline libraries/AP_Beacon/AP_Beacon.cpp:217]:
```suggestion
    if (beacon_instance >= num_beacons || !beacon_state[beacon_instance].healthy) {
```

Even better if it was two lines IMO - might not have had the bug in the first place then.

**Review by peterbarker — APPROVED** [review]
Good find

**Comment by peterbarker** [comment]
Merged, thanks!

**Comment by rmackay9** [comment]
This is included in Copter/Rover-4.1.0-beta7

**Pull request merged**